<?xml version="1.0"?>
<launch>
  <arg name="gdb_debug" default="false"/>
  <arg name="profile" default="false"/>

  <arg name="base_path" default="/home/ken/datasets/kitti"/>
  <arg name="kitti_date" default="2011_10_03"/>
  <arg name="kitti_seq" default="0034"/>

  <arg name="dataset_path" default="$(arg base_path)/$(arg kitti_date)/$(arg kitti_date)_drive_$(arg kitti_seq)_sync"/>

  <arg name="force_gt" default="true"/>

  <group ns="msckf">
    <param name="dataset_path" value="$(arg dataset_path)"/>

    <param name="feature_cov" value="8"/>

    <param name="force_gt_state" value="$(arg force_gt)"/>

    <param name="imu_vars/w_var" value="1e-4"/>
    <param name="imu_vars/dbg_var" value="3.6733e-5"/>
    <param name="imu_vars/a_var" value="1e-2"/>
    <param name="imu_vars/dba_var" value="7e-2"/>

    <param name="imu_covars/p_init" value="1e-12"/>
    <param name="imu_covars/q_init" value="1e-5"/>
    <param name="imu_covars/v_init" value="1e-2"/>
    <param name="imu_covars/ba_init" value="1e-2"/>
    <param name="imu_covars/bg_init" value="1e-2"/>

    <param name="translation_threshold" value="0.05"/><!--Max translation from marginalizing state-->

    <param name="keyframe_transl_dist" value="0.05"/>
    <param name="keyframe_rot_dist" value="0.05"/>

    <param name="min_track_length" value="10"/>
    <param name="max_track_length" value="30"/>
    <param name="max_cam_states" value="40"/>

    <param name="ransac_threshold" value="-1.0"/>

    <param name="n_grid_rows" value="20"/>
    <param name="n_grid_cols" value="60"/>

    <group unless="$(arg profile)">
      <node pkg="msckf_mono" name="kitti_msckf" type="kitti_msckf" unless="$(arg gdb_debug)" output="screen"/>
      <node pkg="msckf_mono" name="kitti_msckf" type="kitti_msckf" launch-prefix="xterm -e gdb --args" if="$(arg gdb_debug)"/>
    </group>                                                     
    <group if="$(arg profile)">                                  
      <node pkg="msckf_mono" name="kitti_msckf" type="kitti_msckf" launch-prefix="perf record -g --call-graph dwarf --output=perf.out.node_name.data --" if="$(arg profile)"/>
    </group>

    <!--node pkg="rviz" name="msckf_rviz" type="rviz" args="-d $(find msckf_mono)/launch/kitti_msckf.rviz"/-->

  </group>
</launch>
